From dcde595b8f189d6cf53c949e893d9caf6afa8979 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Fri, 30 Nov 2012 15:17:01 +0800
Subject: [PATCH 109/310] ENGR00235400 CTS:
 android.bluetooth.cts.BasicAdapterTest*

Fix the android.bluetooth.cts.BasicAdapterTest* CTS failure on SabreSD.

The cause of the failure is we include the
frameworks/native/data/etc/tablet_core_hardware.xml
which have set the bluetooth feature, so the CTS will run BT's case.

The fix was not include this file and add missed feature in
 required_hardware.xml.

Also remove some redundent hardware features.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 imx6/sabresd/SabreSDProductCommon.mk |  6 ++++--
 imx6/sabresd/required_hardware.xml   | 29 ++++++++++-------------------
 2 files changed, 14 insertions(+), 21 deletions(-)

diff --git a/imx6/sabresd/SabreSDProductCommon.mk b/imx6/sabresd/SabreSDProductCommon.mk
index d62f3fa..eecc78f 100644
--- a/imx6/sabresd/SabreSDProductCommon.mk
+++ b/imx6/sabresd/SabreSDProductCommon.mk
@@ -13,14 +13,16 @@ PRODUCT_CHARACTERISTICS := tablet
 PRODUCT_AAPT_CONFIG += xlarge large tvdpi hdpi
 
 PRODUCT_COPY_FILES += \
-	frameworks/native/data/etc/tablet_core_hardware.xml:system/etc/permissions/tablet_core_hardware.xml \
 	frameworks/native/data/etc/android.hardware.location.gps.xml:system/etc/permissions/android.hardware.location.gps.xml \
 	frameworks/native/data/etc/android.hardware.wifi.xml:system/etc/permissions/android.hardware.wifi.xml \
 	frameworks/native/data/etc/android.hardware.wifi.direct.xml:system/etc/permissions/android.hardware.wifi.direct.xml \
 	frameworks/native/data/etc/android.hardware.sensor.light.xml:system/etc/permissions/android.hardware.sensor.light.xml \
 	frameworks/native/data/etc/android.hardware.sensor.accelerometer.xml:system/etc/permissions/android.hardware.sensor.accelerometer.xml \
-	frameworks/native/data/etc/android.hardware.sensor.compass.xml:system/etc/permissions/android.hardware.sensor.compass.xml \
+	frameworks/native/data/etc/android.hardware.telephony.gsm.xml:system/etc/permissions/android.hardware.telephony.gsm.xml \
+	frameworks/native/data/etc/android.hardware.faketouch.xml:system/etc/permissions/android.hardware.faketouch.xml \
+	frameworks/native/data/etc/android.hardware.sensor.barometer.xml:system/etc/permissions/android.hardware.sensor.barometer.xml \
 	frameworks/native/data/etc/android.hardware.camera.front.xml:system/etc/permissions/android.hardware.camera.front.xml \
+	frameworks/native/data/etc/android.hardware.camera.flash-autofocus.xml:system/etc/permissions/android.hardware.camera.flash-autofocus.xml \
 	frameworks/native/data/etc/android.hardware.touchscreen.multitouch.jazzhand.xml:system/etc/permissions/android.hardware.touchscreen.multitouch.jazzhand.xml \
 	frameworks/native/data/etc/android.software.sip.voip.xml:system/etc/permissions/android.software.sip.voip.xml \
 	frameworks/native/data/etc/android.hardware.usb.host.xml:system/etc/permissions/android.hardware.usb.host.xml \
diff --git a/imx6/sabresd/required_hardware.xml b/imx6/sabresd/required_hardware.xml
index c9a2271..0afd91e 100644
--- a/imx6/sabresd/required_hardware.xml
+++ b/imx6/sabresd/required_hardware.xml
@@ -14,33 +14,24 @@
      limitations under the License.
 -->
 
-<!-- These are the hardware components that are included in i.MX reference board
+<!-- XXX: This file is a clone of :frameworks/native/data/etc/tablet_core_hardware.xml file,
+           except the bluetooth is removed and android.software.live_wallpaper was added.
 -->
 <permissions>
-    <feature name="android.hardware.camera" />
-    <feature name="android.hardware.camera.autofocus" />
-    <feature name="android.hardware.camera.flash" />
-    <feature name="android.hardware.camera.front" />
     <feature name="android.hardware.location" />
     <feature name="android.hardware.location.network" />
-    <feature name="android.hardware.location.gps" />
-    <feature name="android.hardware.telephony" />
-    <feature name="android.hardware.telephony.gsm" />
-    <feature name="android.hardware.wifi" />
-    <feature name="android.hardware.wifi.direct" />
-    <feature name="android.hardware.faketouch" />
+    <feature name="android.hardware.sensor.accelerometer" />
+    <feature name="android.hardware.sensor.compass" />
+    <!-- <feature name="android.hardware.bluetooth" /> -->
     <feature name="android.hardware.touchscreen" />
     <feature name="android.hardware.touchscreen.multitouch" />
     <feature name="android.hardware.touchscreen.multitouch.distinct" />
-    <feature name="android.hardware.sensor.accelerometer" />
-    <feature name="android.hardware.sensor.compass" />
-    <feature name="android.hardware.sensor.light" />
-    <feature name="android.hardware.sensor.barometer" />
     <feature name="android.hardware.microphone" />
+    <feature name="android.hardware.screen.portrait" />
     <feature name="android.hardware.screen.landscape" />
-    <feature name="android.hardware.usb.accessory" />
-    <feature name="android.hardware.usb.host" />
+
+    <!-- This is added by FSL. -->
     <feature name="android.software.live_wallpaper" />
-    <feature name="android.software.sip" />
-    <feature name="android.software.sip.voip" />
+
+
 </permissions>
-- 
1.8.0

